File: src/main/java/com/aoranzhang/ezrentback/data/graphql/resolvers/Query.java
package com.aoranzhang.ezrentback.data.graphql.resolvers;

import com.aoranzhang.ezrentback.data.entity.Building;
import com.aoranzhang.ezrentback.data.entity.City;
import com.aoranzhang.ezrentback.data.entity.User;
import com.aoranzhang.ezrentback.service.BuildingService;
import com.aoranzhang.ezrentback.service.CityService;
import com.aoranzhang.ezrentback.service.UserService;
import com.coxautodev.graphql.tools.GraphQLQueryResolver;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import javax.servlet.http.HttpSession;
import java.util.ArrayList;
import java.util.List;
import java.util.Set;

@Component
public class Query implements GraphQLQueryResolver {
    @Autowired
    private HttpSession httpSession;

    @Autowired
    private UserService userService;

    @Autowired
    private CityService cityService;

    @Autowired
    private BuildingService buildingService;

    public User user() {
        String user = (String) httpSession.getAttribute("userEmail");
        if(user != null && user.length() != 0) {
            return userService.getUserByEmail(user);
        }
        return null;
    }

    public Set<City> cities() {
        return cityService.getAllCities();
    }

    public List<String> socialInfo() {
        if(!((Boolean) httpSession.getAttribute("onGoingRegister"))) {
            return null;
        }
        List<String> list = new ArrayList();
        list.add((String) httpSession.getAttribute("newEmail"));
        list.add((String) httpSession.getAttribute("newName"));

        httpSession.setAttribute("onGoingRegister", false);

        return list;
    }

    public Set<Building> buildings(String city) {
        return buildingService.listByCity(cityService.findCityByName(city));
    }

    public Set<Building> properties(String email) {
        return buildingService.listByOwner(userService.getUserByEmail(email));
    }
}

File: src/main/kotlin/com/aoranzhang/ezrentback/data/graphql/resolvers/Query.kt
package com.aoranzhang.ezrentback.data.graphql.resolvers

import com.aoranzhang.ezrentback.data.entity.Building
import com.aoranzhang.ezrentback.data.entity.City
import com.aoranzhang.ezrentback.data.entity.User
import com.aoranzhang.ezrentback.service.BuildingService
import com.aoranzhang.ezrentback.service.CityService
import com.aoranzhang.ezrentback.service.UserService
import com.coxautodev.graphql.tools.GraphQLQueryResolver
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class Query @Autowired constructor(
        private val userService: UserService,
        private val cityService: CityService,
        private val buildingService: BuildingService
) : GraphQLQueryResolver {

    val LOGGER = LoggerFactory.getLogger(Query::class.java)!!

    fun cities(): Set<City> {
        return cityService.allCities
    }

    fun buildings(city: String): Set<Building>? {
        val nullableCity = cityService.findCityByName(city)
        return if (nullableCity == null) {
            null
        }else {
            return buildingService.listByCity(nullableCity)
        }
    }

    fun properties(email: String): Set<Building> {
        val user = userService.getUserByEmail(email)
        return if (user != null) {
            buildingService.listByOwner(user)
        }else {
            HashSet()
        }
    }
}
